contracts/staking/GoodFundManager.sol

```diff
@@ -291,43 +292,49 @@ contract GoodFundManager is DAOUpgradeableContract, DSMath {
 	 * @dev  Function that get addresses of staking contracts that we can collect interest from with a specific gas limit
 	 * but that also pass the interest>=gascosts*multiplier
 	 * @param _maxGasAmount The maximum amount of the gas that keeper willing to spend collect interests
-	 * @return addresses of the staking contracts to the collect interests
+	 * @return addresses of the staking contracts to the collect interests , interests of the staking contracts ,
+	 * Max gas amount can be used for collected interest value,Actual gas required to collect interests
 	 */
 	function calcSortedContracts(uint256 _maxGasAmount)
 		public
 		view
-		returns (address[] memory)
+		returns (
+			address[] memory,
+			uint256[] memory,
+			uint256,
+			uint256
+		)
 	{
-		uint256 activeContractsLength = activeContracts.length;
-		address[] memory addresses = new address[](activeContractsLength);
-		uint256[] memory balances = new uint256[](activeContractsLength);
+		address[] memory addresses = new address[](activeContracts.length);
+		uint256[] memory balances = new uint256[](activeContracts.length);
 		uint256 tempInterest;
 		int256 i;
-		for (i = 0; i < int256(activeContractsLength); i++) {
+		for (i = 0; i < int256(activeContracts.length); i++) {
 			(, , , , tempInterest) = IGoodStaking(activeContracts[uint256(i)])
 				.currentGains(false, true);
 			if (tempInterest != 0) {
 				addresses[uint256(i)] = activeContracts[uint256(i)];
 				balances[uint256(i)] = tempInterest;
 			}
 		}
-		uint256 gasCostInDAI = getGasPriceIncDAIorDAI(_maxGasAmount, true); // Get gas price in DAI so can compare with possible interest amount to get
+		uint256 initialGasAmount = _maxGasAmount;
 		address[] memory emptyArray = new address[](0);
-
+		uint256[] memory emptyUintArray = new uint256[](0);
 		quick(balances, addresses); // sort the values according to interest balance
 		uint256 gasCost;
 		uint256 possibleCollected;
-		for (i = int256(activeContractsLength) - 1; i >= 0; i--) {
+		uint256 maxGasAmount = _maxGasAmount;
+		for (i = int256(activeContracts.length) - 1; i >= 0; i--) {
```

Comment:
this is not what i requested.
there should be no _maxGasAmount input, and we list all contracts. ie we dont break once in loop once maxgas is passed (since there will be no maxgas input)
you go over contracts sorted by interest and you add a line to the output array:
- contract
- interest value in contract
- interest value that will be collected so far including this contract
- actual required gas to collect from all contracts so far including this contract
- max gas that will pass threshold (depending if it is x4 or without x4, ie "interest so far"/4*gasprice)
- bool: true if max gas>=actual required gas

Reply:
@aliberkhsl also noticed, that i've changed the merge PR into the branch you forked from

Reply:
I changed the function according to this description